src/main/java/menu/UserMenu.java

```diff
@@ -64,7 +61,9 @@ public void options() {
 
                         break;
                     case SHOW_CART:
-
+                        cart.refreshCart(new Warehouse());
+                        System.out.println("You have chosen: " + cart.getProductsInCart());
```

Comment:
Devi usare il metodo printDevices() in Researcher.

Reply:
come dice gabriel, va usato il metodo printDevices() in Researcher